verifiers/internal/gha/npm_sigstore_tuf_test.go

```diff
@@ -0,0 +1,120 @@
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+const (
+	testTargetLocalFilePath = "./testdata/registry.npmjs.org_keys.json"
+	testTargetKeyID         = "SHA256:jl3bwswu80PjjokCgh0o2w5c2U4LhQAE57gj9cz1kzA"
+	testTargetKeyUsage      = "npm:attestations"
+	testTargetKeyData       = "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAE1Olb3zMAFFxXKHiIkQO5cJ3Yhl5i6UPp+IhuteBJbuHcA5UogKo0EWtlWwW6KSaKoTNEYL7JlCQiVnkhBktUgg=="
+)
+
+// mockSigstoreTufClient a mock implementation of sigstoreTufClient.
+type mockSigstoreTufClient struct {
+	localPath string
+}
+
+// GetTarget mock implementation of GetTarget for the mockSigstoreTufClient.
+func (client mockSigstoreTufClient) GetTarget(targetPath string) ([]byte, error) {
+	content, err := os.ReadFile(targetPath)
+	if err != nil {
+		return nil, fmt.Errorf("reading mock file: %w", err)
+	}
+	return content, nil
+}
+
+// TestGetNpmjsKeysTarget ensures we can parse the target file.
+func TestGetNpmjsKeysTarget(t *testing.T) {
+	t.Run("parsing local registry.npmjs.org_keys.json", func(t *testing.T) {
+		content, err := os.ReadFile(testTargetLocalFilePath)
+		if err != nil {
+			t.Errorf("reading local file: %s", err)
+		}
+		var expectedKeys npmjsKeysTarget
+		err = json.Unmarshal(content, &expectedKeys)
+		if err != nil {
+			t.Errorf("parsing mock file: %s", err)
+		}
+		mockClient := mockSigstoreTufClient{localPath: testTargetLocalFilePath}
+		actualKeys, err := getNpmjsKeysTarget(mockClient, testTargetLocalFilePath)
+		if err != nil {
+			t.Error(err)
+		}
+		if err != nil {
+			t.Error(err)
+		}
+		if !cmp.Equal(expectedKeys, *actualKeys) {
+			t.Errorf("expected equal values: \nexpected: %v \nactual: %v", expectedKeys, *actualKeys)
+		}
+	})
+
+	t.Run("parsing non-existent registry.npmjs.org_keys.json", func(t *testing.T) {
+		nonExistantPath := "./testdata/my-fake-path"
+		mockClient := mockSigstoreTufClient{localPath: nonExistantPath}
+		_, err := getNpmjsKeysTarget(mockClient, nonExistantPath)
+		if err == nil {
+			t.Error("expected an error")
+		}
+	})
+}
+
+// TestGetKeyDataWithNpmjsKeysTarget ensure that we find the "npm:attestations" key material, given keyid.
+func TestGetKeyDataWithNpmjsKeysTarget(t *testing.T) {
+	tests := []struct {
+		name            string
+		localPath       string
+		keyID           string
+		keyUsage        string
+		expectedKeyData string
+		expectError     bool
+	}{
+		{
+			name:            "npmjs' first attestation key",
+			localPath:       testTargetLocalFilePath,
+			keyID:           testTargetKeyID,
+			keyUsage:        testTargetKeyUsage,
+			expectedKeyData: testTargetKeyData,
+			expectError:     false,
+		},
+		{
+			name:            "missing the 'npm:attestations' keyusage",
+			localPath:       "./testdata/wrong_keyusage_registry.npmjs.org_keys.json",
+			keyID:           testTargetKeyID,
+			keyUsage:        testTargetKeyUsage,
+			expectedKeyData: "", // should not be returned in this error case
+			expectError:     true,
```

Comment:
define errors and match in the unit test instead of using a bool

Reply:
This is essentially the same as my other comment. Though I think it might depend on the types of errors we get from the tuf client. Maybe we could wrap the errors in stable types ourselves if we don't get stable types from the tuf client?

Reply:
Added our own errors to wrap theirs